trace/src/main/org/jetbrains/lincheck/trace/Deserialization.kt

```diff
@@ -169,19 +169,28 @@ private class CodeLocationsContext {
  */
 class LazyTraceReader(
     private val dataFileName: String,
-    private val index: DataInputStream?
+    private val index: DataInputStream?,
+    private val postprocessor: TracePostprocessor
 ) : Closeable {
 
     private fun interface TracepointRegistrator {
         fun register(indexInParent: Int, tracePoint: TRTracePoint, physicalOffset: Long)
     }
 
-    constructor(baseFileName: String) :
+    constructor(baseFileName: String, postprocessor: TracePostprocessor = CompressingPostprocessor) :
             this(
                 dataFileName = baseFileName,
-                index = wrapStream(openExistingFile(baseFileName + INDEX_FILENAME_SUFFIX))
+                index = wrapStream(openExistingFile(baseFileName + INDEX_FILENAME_SUFFIX)),
+                postprocessor = postprocessor
             )
 
+    private fun readTracePointWithPostprocessor(readCallback: () -> TRTracePoint): TRTracePoint =
+        readCallback().let { tracePoint ->
+            if (tracePoint is TRMethodCallTracePoint)
```

Comment:
See the comment about `TracePostprocessor` interface. 
If we change its API to accept just `tracePoint`, this method implementation will become almost trivial: 

```kotlin
postprocessor.postprocess(
    reader = this@LazeTraceReader,
    tracePoint = readCallback(),
)
```